.circleci/config.yml

```diff
@@ -144,6 +144,20 @@ jobs:
       - run:
           name: Deploy hig-react-branch
           command: npm run publish-canary
+  packages-deploy-storybook-on-branch:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependencies-{{ checksum "yarn.lock" }}-{{ checksum "packages/react/yarn.lock" }}-{{ checksum "packages/vanilla/yarn.lock" }}
+      - restore_cache:
+          key: vanilla-build-{{ .Revision }}
+      - restore_cache:
+          key: react-build-{{ .Revision }}
+      - run:
+          name: Deploy Storybook
+          command: npm run storybook-deploy $CIRCLE_BRANCH.surge.sh | awk 'gsub(/\//, "-")'
```

Comment:
It might help to do this work in a script file so you can compute the subdomain in JS.
Might consider name-spacing the domain so we don't crash into other projects.